docs/src/puppeteer-js.md

```diff
@@ -0,0 +1,168 @@
+
+The use of [ElementHandle] is discouraged , use [Locator] objects and web-first assertions instead.
+
+Locators are the central piece of Playwright's auto-waiting and retry-ability. Locators are strict. This means that all operations on locators that imply some target DOM element will throw an exception if more than one element matches given selector.
+
+## Examples
+
+### Automation example
+
+Puppeteer:
+
+```js
+const puppeteer = require('puppeteer');
+
+(async () => {
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+  await page.setViewport({ width: 1280, height: 800 });
+  await page.goto('http://whatsmyuseragent.org/', {
+    waitUntil: 'networkidle2',
+  });
+  await page.screenshot({ path: 'example.png' });
+  await browser.close();
+})();
+```
+
+Line-by-line migration to Playwright:
+
+```js
+const { chromium } = require('playwright'); // 1
+
+(async () => {
+  const browser = await chromium.launch();
+  const page = await browser.newPage(); // 2
+  await page.setViewportSize({ width: 1280, height: 800 }); // 3
+  await page.goto('http://whatsmyuseragent.org/', {
+    waitUntil: 'networkidle', // 4
+  });
+  await page.screenshot({ path: 'example.png' });
+  await browser.close();
+})();
+```
+
+Migration highlights (see inline comments in the Playwright code snippet):
+
+1. Each Playwright Library file has explicit import of `chromium`. Other browsers `webkit` or `firefox` can be used.
+1. For browser state isolation, consider [browser contexts](./browser-contexts.md)
+1. `setViewport` becomes `setViewportSize`
+1. `networkidle2` becomes `networkidle`. Please note that on most cases it is not useful, thanks to auto-waiting.
+
+### Test example
+
+Puppeteer:
+
+```js
+import puppeteer from 'puppeteer';
+
+describe('Playwright homepage', () => {
+  let browser;
+  let page;
+
+  beforeAll(async () => {
+    browser = await puppeteer.launch();
+    page = await browser.newPage();
+  });
+
+  it('contains hero title', async () => {
+    await page.goto('https://playwright.dev/');
+    await page.waitForSelector('.hero__title');
+    const text = await page.$eval('.hero__title', (e) => e.textContent);
+    expect(text).toContain('Playwright enables reliable end-to-end testing');
+  });
+
+  afterAll(() => browser.close());
+});
+```
+Line-by-line migration to Playwright Test:
+
+```js
+import { test, expect } from '@playwright/test'; // 1
+
+test.describe('Playwright homepage', () => {
+  test('contains hero title', async ({ page }) => { // 2, 3
+    await page.goto('https://playwright.dev/');
+    const titleLocator = page.locator('.hero__title'); // 4
+    await expect(titleLocator).toContainText( // 5
+      'Playwright enables reliable end-to-end testing'
+    );
+  });
+});
+```
+
+1. Each Playwright Test file has explicit import of the `test` and `expect` functions
```

Comment:
I think the major distinction would be that in this set up each test is isolated. I.e. if you pollute the localStorage in the first test, second test has clean environment. There are use cases when you don't want that and in those cases you could basically use pptr script as is.

Reply:
I added mention of it line 140, with page fixture to emphasis that you get a new page on every test.